### PR TITLE
[IMP] Forbid other users from deleting messages, add a delete function.

### PR DESCRIPTION
**Pull request by daramousk**

### CODE REVIEW

**Comment by thomaspaulb** [comment]
@hbrunn Could you review this first?

File: mail_edit/models/mail_message.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    @api.multi
+    def check_access_rule(self, operation):
+        if operation in ['write', 'unlink']:
+            try:
+                super(MailMessage, self).check_access_rule(operation)
+            except except_orm:
+                raise
+            else:
+                for rec in self:
+                    if rec.create_uid.id != self.env.uid and \
```
Comment (hbrunn) [inline mail_edit/models/mail_message.py]: I still disagree with this, now with an argument: An admin (someone from group_system, but not root) goes to settings/technical regularly to clean up some mail messages. Don't ask why that's not automated. Maybe it's only fishing out the bosses last late night mail, doesn't matter. But they won't be able to do that any more after installing this module, and there's no way around it like changing access rules or the like, which would be somehow acceptable.

I think that's highly unexpected, and software shouldn't surprise experienced users who have read the manual. So no, please do that on the client side.

File: mail_edit/static/src/js/mail_edit.js [hunk 2]
```diff
@@ -39,6 +40,19 @@ openerp.mail_edit = function (instance) {
                     },
                 });
             });
+        },
+
+        on_message_delete: function (e) {
+            var mail_message_model = new openerp.Model('mail.message');
+            var self = this;
+            mail_message_model.call('unlink', [this.id]).then(
```
Comment (hbrunn) [inline mail_edit/static/src/js/mail_edit.js]: it seems to me checking https://github.com/OCA/OCB/blob/8.0/addons/mail/static/src/js/mail.js#L171 should be enough to do the right thing here already?

Comment (hbrunn) [reply mail_edit/static/src/js/mail_edit.js]: why don't you call https://github.com/OCA/OCB/blob/8.0/addons/mail/static/src/js/mail.js#L1036 anyways? you get a confirmation dialog and if I read the code correctly fancy fadeout for free, then your also don't need the awkward parent assumption (what if the chatter is somewhere else on the form?)

File: mail_edit/static/src/xml/mail_edit.xml [hunk 3]
```diff
@@ -6,6 +6,9 @@
             <span class="oe_edit">
                 <a title="Edit Mail" class="oe_e">&#038;</a>
             </span>
+            <span class="oe_delete">
```
Comment (hbrunn) [inline mail_edit/static/src/xml/mail_edit.xml]: how about ``t-if="widget.is_author``? Or even ``t-att-disabled="widget.is_author None or 'disabled'"`` (from the top off my head, untested). Then there's no need to check for author above

**Review by hbrunn — CHANGES_REQUESTED** [review]

**Comment by daramousk** [comment]
@hbrunn Done

**Review by hbrunn — APPROVED** [review]

**Comment by daramousk** [comment]
@thomaspaulb Anything else you need me to do on this?

**Comment by thomaspaulb** [comment]
@daramousk Nope, looks good, tested and works well also.

**Pull request merged**